### ISSUE TITLE
error: use of value '%54' expects different type than prior uses

### ISSUE THREAD

**Issue by noodlesLJH** [d8ahazard/sd_dreambooth_extension#1197]
## ⚠️If you do not follow the template, your issue may be closed without a response ⚠️
Kindly read and fill this form in its entirety.

### 0. Initial troubleshooting
while training error occurs
loc("/tmp/torchinductor_root/triton/0/034dd918cb3292bd79f7fa0592c06663/triton_mm.ttgir":134:22): error: use of value '%54' expects different type than prior uses: 'tensor<64x64xf32, #triton_gpu.mma<{versionMajor = 1, versionMinor = 0, warpsPerCTA = [4, 1]}>>' vs 'tensor<64x64xf32, #triton_gpu.mma<{versionMajor = 1, versionMinor = 1, warpsPerCTA = [2, 2]}>>'

#######################################################################################################
Python 3.10.10 (main, Mar 21 2023, 18:45:11) [GCC 11.2.0] on linux
Dreambooth revision: 926ae204ef5de17efca2059c334b6098492a0641
SD-WebUI revision: 
[!] xformers NOT installed.
[+] torch version 2.0.0+cu118 installed.
[+] torchvision version 0.15.1+cu118 installed.
[+] accelerate version 0.18.0 installed.
[+] diffusers version 0.14.0 installed.
[+] transformers version 4.26.1 installed.
[+] bitsandbytes version 0.35.4 installed.

#######################################################################################################
```


### 2. Describe the bug

(A clear and concise description of what the bug is)

[db_config.txt](https://github.com/d8ahazard/sd_dreambooth_extension/files/11316252/db_config.txt)


### 3. Provide logs

If a crash has occurred, please provide the *entire* stack trace from the log, including the last few log messages *before* the crash occurred.

```st
Initializing dreambooth training...
The version of diffusers is less than or equal to 0.14.0. Performing monkey-patch...
Pre-processing images: classifiers_0: : 0it [00:00, ?it/s]                             
Nothing to generate.
Found 100 reg images.
Preparing dataset...
Init dataset!
Preparing Dataset (With Caching)
Caching latents...: 100%|██████████| 120/120 [00:02<00:00, 47.68it/s]
Loading cached latents...
Bucket 0 (512, 512, 0) - Instance Images: 20 | Class Images: 100 | Max Examples/batch:  40
Saving cache!
Total Buckets 1 - Instance Images: 20 | Class Images: 100 | Max Examples/batch:  40

Total images / batch: 40, total examples: 40
Total dataset length (steps): 40
Initializing bucket counter!
  ***** Running training *****
  Num batches each epoch = 40
  Num Epochs = 150
  Batch Size Per Device = 1
  Gradient Accumulation steps = 1
  Total train batch size (w. parallel, distributed & accumulation) = 1
  Text Encoder Epochs: 0
  Total optimization steps = 3000
  Total training steps = 6000
  Resuming from checkpoint: False
  First resume epoch: 0
  First resume step: 0
  Lora: True, Optimizer: 8bit AdamW, Prec: fp16
  Gradient Checkpointing: True
  EMA: False
  UNET: True
  Freeze CLIP Normalization Layers: False
  LR: 0.0001
  V2: False
Steps:   0%|          | 0/6000 [00:00<?, ?it/s]loc("/tmp/torchinductor_root/triton/0/034dd918cb3292bd79f7fa0592c06663/triton_mm.ttgir":134:22): error: use of value '%54' expects different type than prior uses: 'tensor<64x64xf32, #triton_gpu.mma<{versionMajor = 1, versionMinor = 0, warpsPerCTA = [4, 1]}>>' vs 'tensor<64x64xf32, #triton_gpu.mma<{versionMajor = 1, versionMinor = 1, warpsPerCTA = [2, 2]}>>'
```

### 4. Environment

What OS?

linux,3.10.0-1160.11.1.el7.x86_64

What GPU are you using?
Tesla V100
NVIDIA-SMI 525.60.13    Driver Version: 525.60.13    CUDA Version: 12.0

**Comment by chouti** [comment]
I've met the same problems here.

**Comment by d8ahazard** [comment]
Try disabling gradient checkpointing?

**Comment by Devilsam12** [comment]
hey i got the same issue. any updates? i tried disabling gradient checkpointing too

**Comment by yqyqyq-W** [comment]
Please help! I got the same issue on Linux.

**Comment by Devilsam12** [comment]
alright i found a fix.
turns out it was a fault with the tensorrt libraries. in particular, if you activated virutal environment and from python tried to import tensorrt, it would show errors, that libcudnn.so files could not be found.
so i copied the whole tensorrt_libs folder and nvidia folder from my local dist-packages and pasted into my venv site packages and the the error persists no more.

**Comment by yqyqyq-W** [comment]
> alright i found a fix. turns out it was a fault with the tensorrt libraries. in particular, if you activated virutal environment and from python tried to import tensorrt, it would show errors, that libcudnn.so files could not be found. so i copied the whole tensorrt_libs folder and nvidia folder from my local dist-packages and pasted into my venv site packages and the the error persists no more.

That doesn't work for me. I can import tensorrt normally.

**Comment by Devilsam12** [comment]
even in your venv?
try activating your venv and importing tensorrt?

**Comment by yqyqyq-W** [comment]
> 

Everthing is tested under runtime environment. Probably tensorrt is not the one to be blamed.
